OpenSim/Simulation/MarkersReference.cpp

```diff
@@ -22,139 +22,157 @@
  * -------------------------------------------------------------------------- */
 
 #include "MarkersReference.h"
-#include <OpenSim/Common/Units.h>
 #include <SimTKcommon/internal/State.h>
+#include <cmath>
 
 using namespace std;
 using namespace SimTK;
 
 namespace OpenSim {
 
-MarkersReference::MarkersReference() : Reference_<SimTK::Vec3>()
-{
+MarkersReference::MarkersReference() :
+    Reference_<SimTK::Vec3>() {
     constructProperties();
     setAuthors("Ajay Seth");
 }
 
 MarkersReference::MarkersReference(const std::string& markerFile,
-    Units modelUnits) : MarkersReference()
-{
+                                   Units modelUnits) :
+    MarkersReference() {
     loadMarkersFile(markerFile, modelUnits);
 }
 
-
-MarkersReference::MarkersReference(MarkerData *markerData,
-    const Set<MarkerWeight>* markerWeightSet) : MarkersReference()
-{
-    _markerData.reset(markerData);
-    if (markerWeightSet!=nullptr) upd_marker_weights()= *markerWeightSet;
-    populateFromMarkerData(*markerData);
+MarkersReference::
+MarkersReference(const TimeSeriesTable_<SimTK::Vec3>& markerTable,
+                 const Set<MarkerWeight>* markerWeightSet,
+                 Units units) :
+    MarkersReference() {
+    _markerTable = markerTable;
+    if(markerWeightSet != nullptr)
+        upd_marker_weights() = *markerWeightSet;
+    populateFromMarkerData(markerTable, units.getAbbreviation());
 }
 
-/* load the marker data for this MarkersReference from markerFile  */
-void MarkersReference::loadMarkersFile(const std::string markerFile, Units modelUnits)
-{
-    upd_marker_file() = markerFile;
-    _markerData.reset(new MarkerData(get_marker_file()));
+void MarkersReference::loadMarkersFile(const std::string markerFile,
+                                       Units modelUnits) {
+    auto fileExt = FileAdapter::findExtension(markerFile);
```

Comment:
Are these case sensitive? We had issues in the past with users providing .TRC, .MOT files, does the function return lowercase regardless? I can look it myself but it's hard to navigate. The test is for ".trc" and the message says "TRC".